source/lac/scalapack.cc

```diff
@@ -323,117 +326,344 @@ void ScaLAPACKMatrix<NumberType>::invert()
 
 
 template <typename NumberType>
-std::vector<NumberType> ScaLAPACKMatrix<NumberType>::eigenvalues_symmetric()
+std::vector<NumberType> ScaLAPACKMatrix<NumberType>::eigenpairs_symmetric_by_index(const std::pair<unsigned int,unsigned int> &index_limits,
+    const bool compute_eigenvectors)
+{
+  // check validity of index limits
+  Assert (index_limits.first != numbers::invalid_unsigned_int,ExcMessage("index limits have invalid value"));
+  Assert (index_limits.second != numbers::invalid_unsigned_int,ExcMessage("index limits have invalid value"));
+  Assert (index_limits.first < (unsigned int)n_rows,ExcMessage("index limits have to be smaller than matrix size"));
```

Comment:
`ExcIndexRange(index_limits.first,0,n_rows)` as it's more telling. Same next line.